code/modules/ghostroles/spawner/human/human.dm

```diff
@@ -159,3 +162,14 @@
 //Proc executed after someone is spawned in
 /datum/ghostspawner/human/post_spawn(mob/user)
 	. = ..()
+	if(visitable_overmap_type)
+		for(var/z_level in map_sectors)
+			if(istype(map_sectors[z_level], visitable_overmap_type))
```

Comment:
Instead of doing this, can't you just get the sector from the mob that just spawned? It'd also handle duplicates better.